Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -475,10 +475,19 @@ DOM::ExceptionOr<void> XMLHttpRequest::send(Optional<XMLHttpRequestBodyInit> bod
     auto request = LoadRequest::create_for_url_on_page(request_url, m_window->page());
     request.set_method(m_method);
     if (body_with_type.has_value()) {
-        body_with_type->body.source().visit(
-            [&](ByteBuffer const& buffer) { request.set_body(buffer); },
-            [&](NonnullRefPtr<FileAPI::Blob> const& blob) { request.set_body(blob->m_byte_buffer); },
-            [](auto&) {});
+        TRY_OR_RETURN_OOM(body_with_type->body.source().visit(
```

Comment:
...and this is why - the `ByteBuffer(ByteBuffer const& other)` ctor hides fallibility :^)